hibernate-testing/src/main/java/org/hibernate/testing/bytecode/enhancement/BytecodeEnhancerRunner.java

```diff
@@ -182,7 +184,7 @@ private static ClassLoader buildEnhancerClassLoader(
 			EnhancementContext enhancerContext,
 			List<EnhancementSelector> selectors) {
 		return new EnhancingClassLoader(
-				Environment.getBytecodeProvider().getEnhancer( enhancerContext ),
+				buildDefaultBytecodeProvider().getEnhancer( enhancerContext ),
```

Comment:
You usually reset caches explicitly at the end, even if the bytecode provider is going to get garbage collected, so I suppose there's a risk of memory leak of classloader pollution or something...

But here you're not doing it? Shouldn't you keep the bytecode provider around to reset caches after the runner finishes "running"?

Reply:
A missing resetCaches() method shouldn't leak the classloader if it's not held in a static field (like it originally was), as it's now held only from the ServiceRegistry instance; AFAIK it's just about freeing up some memory.
I've added the `resetCaches` in other places because it's a saner approach, but it shouldn't be strictly necessary - in fact it wasn't there before.

N.B. the `resetCaches` method didn't exist in the past, I've added it recently to reduce runtime consumption of a bootstrapped Quarkus application.